lib/java/src/org/apache/thrift/TDeserializer.java

```diff
@@ -61,6 +78,54 @@ public TDeserializer(TProtocolFactory protocolFactory) throws TTransportExceptio
     protocol_ = protocolFactory.getProtocol(trans_);
   }
 
+  /**
+   * Construct a new TDeserializer that supports partial deserialization
+   * that outputs instances of type controlled by the given {@code processor}.
+   *
+   * @param thriftClass a TBase derived class.
+   * @param fieldNames list of fields to deserialize.
+   * @param processor the Processor that handles deserialized field values.
+   * @param protocolFactory the Factory to create a protocol.
+   */
+  public TDeserializer(
```

Comment:
(maybe it just works, I'm not familiar with the java codebase and this is not obvious to me)

I don't see you changing any of the existing public functions of `TDeserializer`, for example:
```
public void deserialize​(TBase base, byte[] bytes) throws TException
public void deserialize​(TBase base, byte[] bytes, int offset, int length) throws TException
public void deserialize​(TBase base, java.lang.String data, java.lang.String charset) throws TException
```
Since you are adding new constructors to `TDeserializer`, you need to ensure that those existing public functions still work as intended. When someone uses this constructor to create a `TDeserializer`, then calls `deserialize​(TBase base, byte[] bytes)`, they would expect that `deserialize` function does partial deserialization, not the full one.